.github/workflows/build_and_test_cmake.yml

```diff
@@ -0,0 +1,441 @@
+    runs-on: ${{ matrix.os-version }}
+    strategy:
+      matrix:
+        os-version: [macos-11, macos-12]
+        arch: [x86_64, arm64]
+        config-zstd: [ON, OFF]
+        exclude:
+          # excludes ZSTD on Arm64
+          - arch: arm64
+            config-zstd: ON
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "3.8.x"
+
+      - name: Mark repository as safe (Checkout actions is not run with same UID)
+        run: |
+          git config --global --add safe.directory /__w/PcapPlusPlus/PcapPlusPlus
+          git config --global --add safe.directory ${{ github.workspace }}
+
+      - name: Configure PcapPlusPlus
+        run: |
+          cmake -DLIGHT_PCAPNG_ZSTD=${{ matrix.config-zstd }} -DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} -S . -B $BUILD_DIR
+
+      - name: Build PcapPlusPlus
+        run: cmake --build $BUILD_DIR -j
+
+      - name: Check architecture
+        run: lipo $BUILD_DIR/Pcap++/libPcap++.a -verify_arch ${{ matrix.arch }}
+
+      - name: Install Tcpreplay
+        run: brew install tcpreplay
+
+      - name: Test PcapPlusPlus
+        # We can't run cross compiled binaries
+        if: ${{ matrix.arch == 'x86_64' }}
+        run: |
+          python -m pip install -U pip
+          python -m pip install -r ci/run_tests/requirements.txt
+          python ci/run_tests/run_tests.py --interface en0 --use-sudo
+
+      - name: Test Examples
+        if: ${{ matrix.arch == 'x86_64' }}
+        run: |
+          cd Tests/ExamplesTest
+          python -m pip install -U pip
+          python -m pip install -r requirements.txt
+          python -m pytest --interface en0 --use-sudo --root-path=../../Dist/examples_bin
+
+      - name: Check installation
+        run: cmake --install $BUILD_DIR
+
+      - name: Build Tutorials
+        run: |
+          mkdir -p build_examples
+          cmake -DPCAPPP_BUILD_TUTORIALS=ON -DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} -S Examples -B build_examples
+          cmake --build build_examples -j
+
+      - name: Test Tutorials
+        if: ${{ matrix.arch == 'x86_64' }}
+        run: |
+          # We need to execute the HelloWorld in the same folder as the PCAP file
+          cp build_examples/Tutorials/Tutorial-HelloWorld/Tutorial-HelloWorld Examples/Tutorials/Tutorial-HelloWorld/
+          cd Examples/Tutorials/Tutorial-HelloWorld && ./Tutorial-HelloWorld
+
+  mingw-w64:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        include:
+          - env: i686
+            sys: mingw32
+          - env: x86_64
+            sys: mingw64
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: Setup MSYS2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{matrix.sys}}
+          install: >-
+            git
```

Comment:
isn't git installed by default on the CI machine? 🤔

Reply:
I think it got error without this